mountpoint-s3/src/inode.rs

```diff
@@ -827,9 +835,18 @@ impl SuperblockInner {
                     unreachable!("we know parent is a directory");
                 };
                 if writing_children.contains(&existing_inode.ino()) {
-                    let stat = existing_inode.get_inode_state()?.stat.clone();
+                    let mut sync = existing_inode.get_mut_inode_state()?;
+
+                    let validity = match existing_inode.kind() {
+                        InodeKind::File => self.cache_config.file_ttl,
+                        InodeKind::Directory => self.cache_config.dir_ttl,
+                    };
+                    sync.stat.update_validity(validity);
+                    let stat = sync.stat.clone();
+                    drop(sync);
+
                     Ok(LookedUp {
-                        inode: existing_inode,
+                        inode: existing_inode.clone(),
```

Comment:
We should not need to clone here:
```suggestion
                        inode: existing_inode,
```